File: modifiedfilesdialog.cpp
#include <QTreeWidgetItem>

#include "modifiedfilesdialog.h"
#include "ui_modifiedfilesdialog.h"

ModifiedFilesDialog::ModifiedFilesDialog(QWidget *parent) : QDialog(parent), ui(new Ui::ModifiedFilesDialog) {
	ui->setupUi(this);
}

ModifiedFilesDialog::~ModifiedFilesDialog() {
	delete ui;
}

void ModifiedFilesDialog::on_cancelButton_clicked() {
	this->result = 0;
	this->close();
}

void ModifiedFilesDialog::on_discardAllButton_2_clicked() {
	this->result = 1;
	this->close();
}

void ModifiedFilesDialog::addModifiedItem(SphereFile *item) {
    QTreeWidgetItem *twi = new QTreeWidgetItem(ui->treeWidget);
    //twi->setText(0,item->fileName());
    ui->treeWidget->insertTopLevelItem(0,twi);
}

void ModifiedFilesDialog::on_saveAllButton_clicked() {
	this->result = 2;
	this->close();
}

int ModifiedFilesDialog::getResult() {
	return this->result;
}

File: dialogs/modifiedfilesdialog.py
from PySide6.QtCore import Qt, Slot, Signal
from PySide6.QtWidgets import QDialog, QWidget, QTreeWidgetItem, QAbstractButton, QDialogButtonBox

from ui.ui_modifiedfilesdialog import Ui_ModifiedFilesDialog

from widgets.maintabwidget import MainTabWidget

class ModifiedFilesDialog(QDialog):
	ui: Ui_ModifiedFilesDialog
	paths: list[str]

	@property
	def buttonBox(self):
		return self.ui.buttonBox

	def __init__(self, parent: QWidget):
		super().__init__(parent)
		self.ui = Ui_ModifiedFilesDialog()
		self.ui.setupUi(self)
		self.paths = []

	def clearPaths(self):
		self.ui.treeWidget.clear()
		self.paths.clear()
	
	def addPath(self, path:str):
		item = QTreeWidgetItem(self.ui.treeWidget)
		item.setText(0, path)
		self.paths.append(path)
		self.ui.treeWidget.insertTopLevelItem(0, item)

	def setPaths(self, paths:list[str]):
		self.clearPaths()
		for path in paths:
			self.addPath(path)
